pkg/controller/daemon/daemon_controller.go

```diff
@@ -1175,8 +1175,11 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ds *apps.DaemonSet, nodeL
 
 		if wantToRun {
 			desiredNumberScheduled++
-			if scheduled {
+			if !scheduled {
```

Comment:
I don't follow the logic here. If a pod belonging to a daemonset should be scheduled on a node and it is, then it seems like the `currentNumberScheduled` should be incremented.

Reply:
you may think the ds controller running forever.  currentNumberScheduled  = wantToRun but not exist on every node.  do you agree with this logic?